### ISSUE TITLE
Cache authorization for token in gateway

### ISSUE THREAD

**Issue by korthout** [camunda/camunda#14396]
For each incoming request, the gateway adds authorization to the broker request. The authorization is an encoded JWT containing the authorized tenants. If multi-tenancy is enabled, and identity is configured then the gateway will request the authorized tenants from Identity. Both encoding the JWT and requesting the authorized tenants are costly, which may incur a loss in performance.

To improve the performance, we should cache the authorization of broker requests for a token.

Note that caching should be temporary to avoid sync problems where changes are made to the authorized tenants in Identity.

It would be insightful to set up benchmarks with multi-tenancy and identity before making this change.

**Comment by korthout** [comment]
@koevskinikola With the re-architecture of identity in 8.8, I believe this issue is fixed right? The gateway no longer requests the authorized tenants from identity right?

**Comment by koevskinikola** [comment]
@korthout yes. Zeebe has tenant and authorization information with 8.8, so tenant and authorization checks are performed in the Zeebe Broker (or the service/search layer). No token caching is necessary anymore.

On a side note: I think I already implemented tenant ID caching for as a fix for requests that were timing out, but I'll need some time to find it.

Let's close this issue.
